docs/curl/provenance.md

```diff
@@ -0,0 +1,829 @@
+```
+
+**Output**
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "values": [
+      {
+        "value": "dGhpcyBpcyBhIHRoaXJkIHZhbHVl",
+        "metadata": {
+          "version": {
+            "block_num": 5
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      }
+    ]
+  },
+  "signature": "MEUCIEk90HGFa90Yr/dBDONE+GVD4GaowdQFRL25+S0ECY7tAiEA7+gJSZD8pt0v+KC318xxyLTmLrzcBPnhEMIb/xUmJtI="
+}
+```
+As we can see here, the value returned by query is last value before key was deleted. If we run same query for `key2`, we will get empty list.
+
+### History query for specific version
+Let's check what was value of `key1` at time then tx with position `{"block_num": 6, "tx_num": 0}` was committed.  
+> Actually, this version will be marshalled to `{"block_num":6}`, because `tx_num` equals 0, so we can say it value at block 6 
+
+To do that, we send `/provenance/data/history/{dbname}/{key}?blocknumber={blknum:[0-9]+}&transactionnumber={txnum:[0-9]+}` GET query.
+
+**Sign json query data**
+```sh
+bin/signer -data '{"user_id":"alice","db_name":"db2","key":"key1","version":{"block_num":6}}' -privatekey=deployment/sample/crypto/alice/alice.key
+```
+
+**Signature**
+```
+MEQCIFBH19/vOMSK1Q66djyS6u/G1YSppl24S00VpmsjodYbAiAxzejpgkHRFSKM/lXIw9hC19Tf5JbdSKsJW/SGXPmrdQ==
+```
+
+**Submit query**
+```sh
+curl \
+     -H "Content-Type: application/json" \
+     -H "UserID: alice" \
+     -H "Signature: MEQCIFBH19/vOMSK1Q66djyS6u/G1YSppl24S00VpmsjodYbAiAxzejpgkHRFSKM/lXIw9hC19Tf5JbdSKsJW/SGXPmrdQ==" \
+     -X GET -G "http://127.0.0.1:6001/provenance/data/history/db2/key1" -d blocknumber=6 -d transactionnumber=0 | jq .
+```
+
+> Please note, as mentioned earlier, transaction number eliminated from query used for sign, because it equals to 0, but still exists as part of GET url. In case if transaction index in block is not equal 0, `tx_num` will e part of serialized query to sign. 
+
+**Output**
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "values": [
+      {
+        "value": "dGhpcyBpcyBhIGZpcnN0IHZhbHVlIHVwZGF0ZWQ=",
+        "metadata": {
+          "version": {
+            "block_num": 6
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      }
+    ]
+  },
+  "signature": "MEYCIQC6SFao6V8C3ETjp1AZ6FPZjUVhA8/DG+Vn9gLDTXjsGgIhAKaA3szWlfEZ4D2SDmBLs6KWxlH5t5byplqp6A8gAl4i"
+}
+```
+
+### User related queries
+One of important query user activity query - to know for each user what transaction she submitted, what keys she read or write.  There are three different types of queries for that.
```

Comment:
An important type of query addresses user activity, and allows to know, for a particular user, which transactions she submitted, and which keys she read or wrote. There are three different types of queries for that:
- type 1
- type 2
- type 3 (with links to sections below)

Reply:
done